src/context/WalletProvider/MetaMask/components/Connect.tsx

```diff
@@ -70,14 +65,8 @@ export const MetaMaskConnect = ({ history }: MetaMaskSetupProps) => {
       try {
         const deviceId = await wallet.getDeviceID()
 
-        // Switch to Mainnet if wallet is on any other chain
-        const chainId = await wallet.ethGetChainId?.()
-        if (bnOrZero(chainId).toString() !== CHAIN_REFERENCE.EthereumMainnet) {
-          try {
-            await wallet.ethSwitchChain?.(bn(CHAIN_REFERENCE.EthereumMainnet).toNumber())
-          } catch (e) {
-            throw new Error('walletProvider.metaMask.errors.network')
-          }
+        if (provider?.chainId !== '0x1') {
```

Comment:
Don't need the full revert here, cherry-pick hunks of  f304b4f416a99472525067a96240179fe1cc4855 to still use `ethGetChainId()`, but not the chain switch part. Also remove the wrong network error, we should not need to error anymore or wrong EVM chain since we can just switch it before broadcasting a Tx